### PR TITLE
Create 2014-01-31-stringsandlists.md

### PR DESCRIPTION
**Pull request by bdferr**
This would be longer, as it doesn't include the extra problems, but right now this is the most I can do with the time I have; I started on this project too late. Here is the link to the file on my branch: http://bdferr.github.io/spring2014/2014/01/31/stringsandlists.html

### CODE REVIEW

File: _posts/2014-01-31-stringsandlists.md [hunk 1]
```diff
@@ -0,0 +1,301 @@
...
+  
+print("Your text contains", number_of_letters, "alphabetic characters, of which", number_of_es, "are the letter E.")
+
+```
+
+Here is the output from #3:
+
+```
+
+Your text contains 243 alphabetic characters, of which 109 (44.8%) are 'e'.
```
Comment (eah13) [inline _posts/2014-01-31-stringsandlists.md:67]: Looks like a disconnect here.  I'm not sure how this output could possibly come from the code above as it's written.

File: _posts/2014-01-31-stringsandlists.md [hunk 2]
```diff
@@ -0,0 +1,301 @@
...
+        new_string = word + " " + new_string
+    return new_string
+print(reverse_by_word("Son I am able she said / though you scare me watch said I / beloved I said watch me scare you though / said she able am I son"))
+
+```
+
+7.
+Am I allowed to copy the code which was already in the textbook as the answer to this problem? 
+It was surprisingly concise and I don't see how I could improve upon it, except to try different words
+with the "testEqual" function. All the tests yielded "Pass".
```
Comment (eah13) [inline _posts/2014-01-31-stringsandlists.md:102]: Good question.  Copying the code without thinking about it is unlikely to deepen your understanding.  That said, if you've attempted the problem then see how the book's answer is better, just describe this in your post.

File: _posts/2014-01-31-stringsandlists.md [hunk 3]
```diff
@@ -0,0 +1,301 @@
...
+testEqual(remove('an', 'banana'),'bana')
+testEqual(remove('cyc', 'bicycle'), 'bile')
+testEqual(remove('iss', 'Mississippi'), 'Missippi')
+
+```
+
+12. 
+Here is exercise #12 for the strings chapter; it does not quite get unlimited instances
+of a substring out of a string, but I did figure out how to take two instances out,
+which was something, at least.
```
Comment (eah13) [inline _posts/2014-01-31-stringsandlists.md:147]: Good catch.  Recursion or a while loop that modifies the string as you go are two strategies that would solve this.

File: _posts/2014-01-31-stringsandlists.md [hunk 4]
```diff
@@ -0,0 +1,301 @@
...
+The output from #9 was -15.
+
+14:
+I don't see what this has to do with the split and join methods,
+although if I were to try to replace occurences of a particular word
+rather than a character, I would use the "split" function.
+It would be easier, though, to make it replace particular strings
+rather than words.
+In this version I am making it replace particular characters, which was
+also exercise #8 in the strings chapter.
```
Comment (eah13) [inline _posts/2014-01-31-stringsandlists.md:228]: Good try here.  A problem is that your code works only for one character long replacements.  If you'd used the 'split' method you'd end up with a list of characters that you could just insert the new string in between.

**Comment by eah13** [comment]
@bdferr nice work here, even though you didn't get to the extra exercises.  Code early and often!

File: _posts/2014-01-31-stringsandlists.md (hunk 1 above)
Comment (bdferr) [inline _posts/2014-01-31-stringsandlists.md:67]: You are right; I seem to have copied output from a version which actually calculated the percentage which I don't remember ever even writing. I can't figure out where that came from as I don't have a copy of it anywhere. I can guess that it included assigning some variable, let's say x, to (number_of_es / number_of_letters \* 100) and then printing "...of which", number_of_letters, "%s%" %x, "are 'e'." I would test this right now, but interactivepython.org is not responding.

**Pull request merged**